### PR TITLE
changes in tensor layer

### PR DESCRIPTION
**Pull request by d-netto**

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
The test error looks real

**Comment by d-netto** [comment]
Dimensions were inverted.

File: Project.toml [hunk 1]
```diff
@@ -20,6 +22,7 @@ ProgressLogging = "33c8b6b6-d38a-422a-b730-caa89a2f386c"
 RecursiveArrayTools = "731186ca-8d62-57ce-b412-fbd966d074cd"
 Requires = "ae029012-a4dd-5104-9daa-d747884805df"
 ReverseDiff = "37e2e3b7-166d-5795-8a7a-e32c996b4267"
+Revise = "295af30f-e4ad-537b-8983-00126c2a3abe"
```
Comment (Vaibhavdixit02) [inline Project.toml]: This shouldn't be added as a dependency.

Comment (d-netto) [reply Project.toml]: yeah, I believe it's fixed in the latest commit

Comment (Vaibhavdixit02) [reply Project.toml]: My bad, sorry

**Comment by d-netto** [comment]
Tests passed. Travis outputting "in progress" for some reason.

**Comment by ChrisRackauckas** [comment]
Great. I just realized... can we test whether this works with minibatching? I am not sure it does.

**Comment by d-netto** [comment]
Ok. I'll create a test for mini-batching.

**Pull request merged**